homeassistant/components/xiaomi_miio/vacuum.py

```diff
@@ -73,6 +73,8 @@
     17: STATE_CLEANING,  # "Zoned cleaning"
     18: STATE_CLEANING,  # "Segment cleaning"
     22: STATE_DOCKED,  # "Emptying the bin" on s7+
+    23: STATE_DOCKED,  # "Washing the mop" on s7maxV
+    26: STATE_DOCKED,  # "Going to wash the mop" on s7maxV
```

Comment:
Should this be returning? Not that it matters much, just wondering..

Reply:
That is a very good question, I do not know since I have the auto-empty dock instead of the ultra dock.
@ihaettypo Do you know what the S7MaxV is doing when it is "Going to wash the mop", is it already docked and starting up the washing procedure, or is it not docked and returning to the dock to get the mop washed?

Reply:
I changed it not to Returning since I guess that is more likely, but lets wait for @ihaettypo response.

Reply:
I was going to add the milestone, but maybe it should be added after the response from @ihaettypo.

Reply:
Hi thanks for fixing this and sorry for the delay, I was on a business trip. I just checked, when "Going to wash the mop" it's actually still on the way to the dock, only when it is fully docked (with the tail inside the dock not the face tho) is it "Washing the mop".

However there's the case that when it is originally docked, and you ask it to start working, it will wash the mop first, which emits "Going to wash the mop" first then "Washing the mop" then "Cleaning". In this case, maybe it's not 100% accurate to call it "returning" since what it does is just moving out right in front of the dock, turning around (where "Going to wash the mop" happens), then backing its tail inside the dock to wash the map. But maybe it's not that far-fetched either.

Reply:
@ihaettypo thanks for the detailed response!
Sounds good, unfortunately HomeAssistant only supports the states:
STATE_CLEANING, STATE_DOCKED, STATE_PAUSED, STATE_IDLE, STATE_RETURNING and STATE_ERROR.
So I think the STATE_RETURNING is the best match for "Going to wash the mop".